extension.js

```diff
@@ -41,104 +73,337 @@ export default class PipOnTop extends Extension
+    try {
       window._isPipAble = null;
+    } catch (e) {
+      // ignore
+    }
+  }
+
+  _cleanupNotifyId(window) {
+    try {
+      if (!window) return;
+      if (window._notifyPipTitleId) {
+        try {
+          window.disconnect(window._notifyPipTitleId);
+        } catch (e) {
+          // disconnect may fail if already disconnected - swallow
+        }
+        window._notifyPipTitleId = null;
+      }
+      // clear any stored wrapper
+      if (window._notifyPipHandler) {
+        window._notifyPipHandler = null;
+      }
+      // cancel scheduled idle if any
+      if (window._pipIdleId) {
+        try {
+          GLib.source_remove(window._pipIdleId);
+        } catch (e) {
+          // ignore
+        }
+        window._pipIdleId = null;
+      }
+    } catch (e) {
+      logError(e, 'PiP on Top: _cleanupNotifyId error');
+    }
+  }
+
+  _scheduleCheck(window) {
+    if (!window) return;
+
+    // if a check is already scheduled, keep it (prevents flooding)
+    if (window._pipIdleId) return;
+
+    try {
+      // schedule a short timeout to avoid racing window destruction
+      window._pipIdleId = GLib.timeout_add(GLib.PRIORITY_DEFAULT, CHECK_DELAY_MS, () => {
+        // clear scheduled id immediately
+        try {
+          window._pipIdleId = null;
+        } catch (e) {
+          // ignore
+        }
+
+        if (!window) return GLib.SOURCE_REMOVE;
+        if (window._destroyed) {
+          // clean up listeners for destroyed window
+          this._cleanupNotifyId(window);
+          return GLib.SOURCE_REMOVE;
+        }
+
+        try {
+          this._checkTitle(window);
+        } catch (e) {
+          logError(e, 'PiP on Top: scheduled _checkTitle error');
+        }
+        return GLib.SOURCE_REMOVE;
+      });
+    } catch (e) {
+      // if scheduling failed, run synchronously but guarded
+      try {
+        this._checkTitle(window);
+      } catch (err) {
+        logError(err, 'PiP on Top: fallback _checkTitle error');
+      }
+    }
+  }
+
+  _isWindowTitlePipLike(title) {
+    if (!title) return false;
+
+    try {
+      if (title === 'Picture-in-Picture') return true;
+      if (title === _('Picture-in-Picture')) return true;
+      if (title === 'Picture in picture') return true;
+      if (title === 'Picture-in-picture') return true;
+      if (title.endsWith && title.endsWith(' - PiP')) return true;
+      if (title === 'TelegramDesktop') return true;
+    } catch (e) {
+      // title could be weird; treat as not PiP
+    }
+    return false;
   }
 
-  _checkTitle(window)
-  {
-    if (!window.title)
+  _safeMakeAbove(window) {
+    if (!window) return;
+    if (window._destroyed) return;
+
+    try {
+      if (typeof window.make_above === 'function') {
+        window.make_above();
+      }
+    } catch (e) {
+      logError(e, 'PiP on Top: _safeMakeAbove error');
+    }
+  }
+
+  _safeUnmakeAbove(window) {
+    if (!window) return;
+    if (window._destroyed) return;
+
```

Comment:
I think most of those checks can be writen in a "simpler" way using  the Optional Chaining Operator (?.)
```suggestion
    if (!window?._destroyed) return;

```